script.js

```diff
@@ -1,12 +1,261 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+// Cache for fetched data
+const cache = {
+  shows: null,
+  episodes: {},
+  firstFetch: false,
+};
+
+let fetchedEpisodes = null;
+
+// Fetch shows data
+function fetchShows() {
+  if (cache.shows) {
+    populateShowDropdown(cache.shows);
+  } else {
+    fetch("https://api.tvmaze.com/shows")
+      .then((response) => response.json())
+      .then((data) => {
+        cache.shows = data;
+        populateShowDropdown(cache.shows);
+        displayAllShows(cache.shows);
+      })
+      .catch((error) => console.error("Error fetching shows:", error));
+  }
+}
+
+function displayAllShows(shows) {
+  const root = document.getElementById("root");
+  root.innerHTML = "";
+  shows.map((show) => displayShows(show));
+}
+
+/// Displaying shows
+function displayShows(show) {
+  const root = document.getElementById("root");
+
+  const showImg = document.createElement("img");
+  showImg.src = show.image.medium ?? "";
+  showImg.alt = show.name;
+
+  const showName = document.createElement("h2");
+  showName.innerHTML = "Show Name: " + show.name;
+
+  const showSummary = document.createElement("p");
+  showSummary.innerHTML = "Summary: " + show.summary;
+
+  const cardShow = document.createElement("div");
+  cardShow.className = "mainShowCard";
+
```

Comment:
Considering that each `cardShow` represents a list of elements, using a `<ul>` for the container -`root` and `<li>` for each card -`cardShow` would offer a more semantical  structure.